Build fails if some struct fields are cfg'd out

If some of the struct's fields are `#[cfg()]` gated and the condition evaluates to false, then the build will fail because atomic_enum's generated code ignores the `#[cfg()]` statements.  For example,
```rust
#[atomic_enum]
enum MyEnum {
    Foo,
    #[cfg(target_os = "multics")]
    Bar,
    #[cfg(not(target_os = "multics"))]
    Baz
}
```
This file will fail like this:
```
error[E0599]: no variant or associated item named `Bar` found for enum `MyEnum` in the current scope
  --> tests/cfg.rs:11:5
   |
7  |   #[atomic_enum]
   |   -------------- variant or associated item `Bar` not found for this enum
8  |   enum MyEnum {
   |  ______-
9  | |     Foo,
10 | |     #[cfg(target_os = "multics")]
11 | |     Bar,
   | |    -^^^
   | |    ||
   | |    |variant or associated item not found in `MyEnum`
   | |____|help: there is a variant with a similar name: `Baz`
```
Examining the generated code with `cargo expand` shows code that wrongly includes `Bar`, like this:
```rust
    fn from_usize(val: usize) -> MyEnum {
        #![allow(non_upper_case_globals)]
        const USIZE_Foo: usize = MyEnum::Foo as usize;
        const USIZE_Bar: usize = MyEnum::Bar as usize;
        const USIZE_Baz: usize = MyEnum::Baz as usize;
        match val {
            USIZE_Foo => MyEnum::Foo,
            USIZE_Bar => MyEnum::Bar,
            USIZE_Baz => MyEnum::Baz,
            _ => {
                ::core::panicking::panic_fmt(format_args!("Invalid enum discriminant"));
            }
        }
    }
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -93,20 +93,19 @@ fn enum_to_usize(ident: &Ident) -> TokenStream2 {
 fn enum_from_usize(ident: &Ident, variants: impl IntoIterator<Item = Variant>) -> TokenStream2 {
     let variants_with_const_names: Vec<_> = variants
         .into_iter()
-        .map(|v| v.ident)
-        .map(|id| {
-            let c_id = Ident::new(&format!("USIZE_{}", &id), id.span());
-            (id, c_id)
+        .map(|v| {
+            let c_id = Ident::new(&format!("USIZE_{}", &v.ident), v.ident.span());
+            (v.attrs, v.ident, c_id)
         })
         .collect();
 
     let variant_consts = variants_with_const_names
         .iter()
-        .map(|(id, c_id)| quote! { const #c_id: usize = #ident::#id as usize; });
+        .map(|(attrs, id, c_id)| quote! { #(#attrs)* const #c_id: usize = #ident::#id as usize; });
 
     let variants_back = variants_with_const_names
         .iter()
-        .map(|(id, c_id)| quote! { #c_id => #ident::#id, });
+        .map(|(attrs, id, c_id)| quote! { #(#attrs)* #c_id => #ident::#id, });
 
     quote! {
         fn from_usize(val: usize) -> #ident {

--- a/tests/cfg.rs
+++ b/tests/cfg.rs
@@ -1,0 +1,20 @@
+#![allow(unexpected_cfgs)] // multics is deliberately always false
+
+use atomic_enum::atomic_enum;
+
+#[atomic_enum]
+enum MyEnum {
+    Foo,
+    #[cfg(target_os = "multics")]
+    Bar,
+    #[cfg(not(target_os = "multics"))]
+    Baz
+}
+
+// Foo and Baz should both be constructible.  Bar should not be, but that can only be verified from
+// a doc test.
+#[test]
+fn construction() {
+    let _ = AtomicMyEnum::new(MyEnum::Foo);
+    let _ = AtomicMyEnum::new(MyEnum::Baz);
+}

